Code/Core/Process/Process.cpp

```diff
@@ -464,16 +466,21 @@ bool Process::IsRunning() const
 
 // WaitForExit
 //------------------------------------------------------------------------------
-int32_t Process::WaitForExit()
+Process::ExitReason Process::WaitForExit(int32_t & exitCodeOut)
 {
     ASSERT( m_Started );
     m_Started = false;
 
-    #if defined( __WINDOWS__ )
+    if ( m_ExitReason == PROCESS_EXIT_TIMEOUT ||
+         m_ExitReason == PROCESS_EXIT_TIMEOUT_INACTIVE )
+    {
+        return m_ExitReason;
```

Comment:
This path does not set `exitCodeOut` before returning.

Is that intentional? If so, document when it will be set and when it won't (e.g. "`exitCodeOut` is only valid when the returned `ExitReason` is...`.

I'd expect that `exitCodeOut` would always be set by the time the function returns.

Reply:
My original intent is that `exitCodeOut` would only be valid in the case that the process exited normally. I should document that.